src/udp.cpp

```diff
@@ -125,7 +129,16 @@ namespace oxen::quic
                 ev_,
                 sock_,
                 EV_READ | EV_PERSIST,
-                [](evutil_socket_t, short, void* self) { static_cast<UDPSocket*>(self)->receive(); },
+                [](evutil_socket_t, short, void* self_) {
+                    auto& self = *static_cast<UDPSocket*>(self_);
+                    self.receive();
+                    if (self.pending_receive_batch_)
+                    {
+                        self.pending_receive_batch_ = false;
+                        if (self.receive_callback_batch_)
+                            self.receive_callback_batch_();
+                    }
```

Comment:
I'm wondering about this boolean (`self.pending_receive_batch_`). Few questions/thoughts we can chat about later:

1) I feel like there shouldn't be a case where `self.pending_receive_batch_` would be true and `self.receive_callback_batch_` would not be set? If the user doesn't provide a batch processing cb, I would guess that the batch processing logic should be avoided entirely? Though that boolean is flipped to true at line 206 `::process_packet`, implying its flipped and checked invariant of whether batch processing is "active" for this endpoint or not
2) What if we used the set-ness of the callback as an indicator of whether batch processing is "active" or "inactive" for this endpoint? Then the boolean could be a "we have a full batch" signal of sorts
3) We could wrap the callback in a lightweight struct that has two members, a callback and an integer/size_t specifying the size of the batch? The size parameter could even default to something sensible at or below the loop limit of 64 packets

Follow-on... would be cool to speed-test this with different batch sizes to see what works most optimally.

Follow-on follow-on... maybe even approximate grid search cross-validation across all our configurable parameters...